src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.cs

```diff
@@ -427,6 +427,7 @@ private static void AssertPlaceholderReplacement(BoundValuePlaceholderBase place
             Debug.Assert(value.Type is { } && (value.Type.Equals(placeholder.Type, TypeCompareKind.AllIgnoreOptions) || value.HasErrors));
         }
 
+#if DEBUG
```

Comment:
This shouldn't be needed. If it's needed, that should be fixed in the analyzer. The analyzer shouldn't fire in a certain configuration for a method if the method is conditional and doesn't apply to that configuration.

Reply:
it looks like it can be removed if the callsite removed `#if DEBUG` too. I can make that update along with fixing the current failure :(

Reply:
If this is deemed to be an analyzer bug, I wouldn't make a further effort than what you have to work around it here, unless the change is very straightforward.

Let's just make sure we've filed the bug in the appropriate place. It would be good to add a comment linking to the bug wherever our workaround is (in this commit, it would be above the `#if DEBUG` here.)

Reply:
> it looks like it can be removed if the callsite removed `#if DEBUG` too. I can make that update along with fixing the current failure (which feels a bit tricky to me :( )

Yes, but that seems like a workaround too. I feel like the analyzer should understand `Conditional` and not mark the member as unused if it's not even compiled.